lib/surface/compiler.ex

```diff
@@ -262,11 +262,30 @@ defmodule Surface.Compiler do
   defp node_type(_), do: :text
 
   defp process_directives(%{directives: directives} = node) when is_list(directives) do
+    {directives, _} =
+      for %AST.Directive{module: mod, meta: meta} = directive <- directives,
+          function_exported?(mod, :process, 2),
+          reduce: {node, MapSet.new()} do
+        {node, processed_directives} ->
+          if MapSet.member?(processed_directives, directive.name) and match?(%AST.Tag{}, node) do
+            message = """
+            the directive `:on-#{directive.name}` has been passed multiple times. Considering only the last value.
```

Comment:
What if the directive is not an `:on-*`? E.g. `:if`. We should probably have a test for this case too.